Makefile.common

```diff
@@ -34,7 +34,14 @@ CPPFLAGS = -Wall -Werror -DDEBUG -I$(TOPDIR)/include
 LDFLAGS  = -nostdlib -T $(TOPDIR)/mips/malta.ld
 
 LDLIBS += -L$(TOPDIR)/sys -L$(TOPDIR)/mips -L$(TOPDIR)/stdc \
-	  -Wl,--start-group -lsys -lmips -lstdc -lgcc -Wl,--end-group
+	  -Wl,--start-group \
+        -Wl,--whole-archive \
+          -lsys \
```

Comment:
`-lsys` must be linked as `--whole-archive`. Otherwise linker set entries declared in translation units that do not use the linker set are stripped from the final kernel image, because linker notices that these symbols are never referenced and therefore do not need to be provided in the final binary. Therefore set entries declared in the same translation unit as the one that is iterating over the set would be the only ones present. In order to prevent the linker from emptying `linker_set_*` sections, I ask it to provide the entire `-lsys` in the final image. There is no linker flag to fine-tune this behavior (e.g. ask to provide all symbols from a specific section).